### PR TITLE
pik: init at 0.9.0

### PR DESCRIPTION
**Pull request by bew**
https://github.com/jacek-kurlit/pik

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/pik/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    rev = version;
+    hash = "sha256-YAnMSVQu/E+OyhHX3vugfBocyi++aGwG9vF6zL8T2RU=";
+  };
+
+  cargoHash = "sha256-a7mqtxZMJl8zR8oCfuGNAiT5MEAmNpbDLSgi8A6FfPA=";
+
+  passthru.tests.version = testers.testVersion { package = pik; };
+
+  meta = {
+    description = "Process Interactive Kill";
```
Comment (FliegendeWurst) [inline pkgs/by-name/pi/pik/package.nix:25]:
```suggestion
    description = "Interactive process killer";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
The Mac build fails with:
```
note: ld: framework not found IOKit
note: ld: library not found for -lobjc
```
Could you fix it or mark meta.platforms = linux?

File: pkgs/by-name/pi/pik/package.nix (hunk 1 above)
Comment (bew) [reply pkgs/by-name/pi/pik/package.nix:25]: Well I agree it's not great english, but it's the description on the repo so I thought it made more sense, and it kinda explains why it's named `pik` `¯\_(ツ)_/¯`

**Comment by bew** [comment]
> The Mac build fails with:
> 
> ```
> note: ld: framework not found IOKit
> note: ld: library not found for -lobjc
> ```
> 
> Could you fix it or mark meta.platforms = linux?

I'll try to fix during the week, as I'm now trying the MacOS platform at work (with Nix obviously)

**Comment by Aleksanaa** [comment]
It's no longer needed as those apple_sdk components are changed to be included by default recently

**Comment by Aleksanaa** [comment]
@ofborg build pik

**Comment by bew** [comment]
Good to know! Do I need to rebase first to get the new behavior or is the build process pre-merging before build?

**Comment by Aleksanaa** [comment]
> Good to know! Do I need to rebase first to get the new behavior or is the build process pre-merging before build?

Still no need to.

**Pull request merged**